extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/ForwardedParser.java

```diff
@@ -131,60 +135,28 @@ private void calculate() {
 
         boolean isProxyAllowed = trustedProxyCheck.isProxyAllowed();
         if (isProxyAllowed) {
-            String forwarded = delegate.getHeader(FORWARDED);
-            if (forwardingProxyOptions.allowForwarded && forwarded != null) {
-                Matcher matcher = FORWARDED_PROTO_PATTERN.matcher(forwarded);
-                if (matcher.find()) {
-                    scheme = (matcher.group(1).trim());
-                    port = -1;
+            if (forwardingProxyOptions.allowForwarded && forwardingProxyOptions.allowXForwarded) {
+                Forwarded forwardedHeaders = null;
+                Forwarded xForwardedHeaders = null;
+                if (ProxyConfig.ForwardedPrecedence.FORWARDED == forwardingProxyOptions.forwardedPrecedence) {
+                    // Forwarded values may override X-Forwarded values if strict forwarded control is disabled
+                    xForwardedHeaders = processXForwarded();
+                    forwardedHeaders = processForwarded();
+                } else {
+                    // X-Forwarded values may override Forwarded values if strict forwarded control is disabled
+                    forwardedHeaders = processForwarded();
+                    xForwardedHeaders = processXForwarded();
                 }
-
-                matcher = FORWARDED_HOST_PATTERN.matcher(forwarded);
-                if (matcher.find()) {
-                    setHostAndPort(matcher.group(1).trim(), port);
-                }
-
-                matcher = FORWARDED_FOR_PATTERN.matcher(forwarded);
-                if (matcher.find()) {
-                    remoteAddress = parseFor(matcher.group(1).trim(), remoteAddress != null ? remoteAddress.port() : port);
+                if (forwardingProxyOptions.strictForwardedControl
+                        && !xForwardedHeaders.modifiedPropertiesMatch(forwardedHeaders)) {
+                    delegate.response().setStatusCode(400);
```

Comment:
Can you add a log message here? Probably `debug`. I know we will have questions - that would help us understand what's going on.

Reply:
Ok, there is a debug statement in the `modifiedPropertiesMatch` method indicating which key does not match. So, it's all good.

Reply:
@cescoffier Sure, I also added some messages here as well to inform that the check is done due to the strict forwarded control